libs/deal-ticket/src/components/deal-ticket/time-in-force-selector.tsx

```diff
@@ -28,11 +29,23 @@ export const timeInForceLabel = (tif: string) => {
   }
 };
 
+type PossibleOrderKeys = Exclude<
+  Schema.OrderType,
+  Schema.OrderType.TYPE_NETWORK
+>;
+type PrevSelectedValue = {
+  [key in PossibleOrderKeys]: Schema.OrderTimeInForce;
+};
+
 export const TimeInForceSelector = ({
   value,
   orderType,
   onSelect,
 }: TimeInForceSelectorProps) => {
+  const [prevValue, setPrevValue] = useState<PrevSelectedValue>({
```

Comment:
suggestion (non blocking): I'm fine with this as is, but I think a better solution would be lift this state into the parent, and add keep the TIF logic in the on click handler of the type selector

Reply:
We would need to add some setter for ref value or pass ref down to the tif selector. Not quite sure it could be for us a big adventage...